File: planning-idsdp-rms-frontend/src/app/main/modules/training-institute/services/progress-verification.service.ts
import {Injectable} from '@angular/core';
import {environment} from "../../../../../environments/environment";
import {HttpClient} from "@angular/common/http";
import {ProgressVerificationRequest} from "../models/progress-verification.request";
import {ProgressVerificationModel} from "../models/progress-verification.model";
import {PaginationResponse} from "../models/pagination-response.model";

@Injectable({
    providedIn: 'root'
})
export class ProgressVerificationService {
    baseTiEndPoint = environment.ibcs.tiBackend;

    constructor(private http: HttpClient) {
    }


    addProgressVerification(progressVerificationModel: ProgressVerificationRequest) {
        return this.http.post(this.baseTiEndPoint + "progress-verification", progressVerificationModel);
    }

    getProgressVerificationList(page: number, size: number) {
        return this.http.get<PaginationResponse<ProgressVerificationModel[]>>(this.baseTiEndPoint + "progress-verification" + "?page=" + page + "&size=" + size);
    }

    getProgressVerificationById(progressVerificationId: number) {
        return this.http.get<ProgressVerificationModel>(this.baseTiEndPoint + "progress-verification/" + progressVerificationId);
    }

    editProgressVerfication(progressVerificationRequest: ProgressVerificationRequest, progressVerificationId: number) {
        return this.http.put(this.baseTiEndPoint + "progress-verification/" + progressVerificationId, progressVerificationRequest);
    }

    deleteProgressVerification(progressVerificationId: number) {
        return this.http.delete(this.baseTiEndPoint + "progress-verification/" + progressVerificationId);
    }
}

File: planning-idsdp-rms-ti-backend/src/main/java/com/ibcs/idsdp/trainninginstitute/services/ProgressVerificationService.java
package com.ibcs.idsdp.trainninginstitute.services;

import com.ibcs.idsdp.trainninginstitute.model.domain.ProgressVerificationModel;
import com.ibcs.idsdp.trainninginstitute.web.dto.request.ProgressVerificationRequest;
import com.ibcs.idsdp.trainninginstitute.web.dto.response.ApiMessageResponse;
import com.ibcs.idsdp.trainninginstitute.web.dto.response.PaginationResponse;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface ProgressVerificationService {
    ResponseEntity<ApiMessageResponse> createProgressVerification(ProgressVerificationRequest progressVerificationRequest);

    ResponseEntity<PaginationResponse<List<ProgressVerificationModel>>> getProgressVerification(int page, int size);

    ResponseEntity<ApiMessageResponse> editProgressVerification(Long progressVerificationId, ProgressVerificationRequest progressVerificationRequest);

    ResponseEntity<ApiMessageResponse> deleteProgressVerification(Long progressVerificationId);

    ResponseEntity<ProgressVerificationModel> getProgressVerificationById(Long progressVerificationId);
}
